lib/output.js

```diff
@@ -74,9 +74,12 @@ function OutputStream(options) {
     var OUTPUT = "";
 
     function to_ascii(str, identifier) {
-        return str.replace(/[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
-            var code = ch.charCodeAt(0).toString(16);
-            if (code.length <= 2 && !identifier) {
+        return str.replace(/[\ud800-\udbff][\udc00-\udfff]|[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
```

Comment:
https://en.wikipedia.org/wiki/UTF-16#U.2B10000_to_U.2B10FFFF (yeah, it's not easy stuff unless someone cares about encoding)